metatomic-torch/include/metatomic/torch/misc.hpp

```diff
@@ -3,13 +3,22 @@
 
 #include <string>
 
+#include <torch/script.h>
+
 #include "metatomic/torch/exports.h"
 
 namespace metatomic_torch {
 
+
 /// Get the runtime version of metatensor-torch as a string
 METATOMIC_TORCH_EXPORT std::string version();
 
+/// Select device on 'supported_devices', a 'desired_device' and what's available on the current machine.
+METATOMIC_TORCH_EXPORT torch::Device pick_device(
+	const std::vector<std::string>& supported_devices,
+	const std::string& desired_device = ""
```

Comment:
```suggestion
	torch::optional<std::string> desired_device = torch::nullopt
```

This should be a bit nicer